LXStudio-IDE/src/main/java/titanicsend/pattern/PeriodicPattern.java

```diff
@@ -0,0 +1,45 @@
+package titanicsend.pattern;
+
+import java.util.*;
+import java.util.concurrent.Callable;
+
+import heronarts.lx.LX;
+import heronarts.lx.LXCategory;
+
+@LXCategory("Testahedron")
+public abstract class PeriodicPattern extends TEPattern {
```

Comment:
note: I think we should use LX's  where possible to drive periodic behavior in patterns. This seems roughly equivalent to the [Click](https://lx.studio/api/heronarts/lx/modulator/Click.html) modulator for example. I'm not sure at first glance how well that would work for the EdgeRunner use case, however.

Reply:
Great find. I'll switch over to that in my next PR.

Reply:
Tried this out; Click turns out to be not as good. Consider the case where more than one period has passed since the last loop. PeriodicPattern is able to tell the loop how many periods it missed so it can catch up. Click can't; it's only able to tell the loop that at least one period has passed.